### PR TITLE
Doc update

### LINKED ISSUE #2547: Update mac installlation instruction to tell how to use clang.

### PR DESCRIPTION
**Pull request by abergeron**
This mainly cleans up the Mac install instructions but there are some other fixes.

Closes #2547.

### CODE REVIEW

File: doc/install.txt [hunk 1]
```diff
@@ -507,62 +507,38 @@ Note that:
...
 
         $ sudo pip install Theano
 
-If you want the bleeding edge version, `download
-and install git <http://git-scm.com/downloads>`_. Then in a
-terminal excute this command:
-
 .. code-block:: bash
 
     $ sudo pip install --upgrade --no-deps git+git://github.com/Theano/Theano.git
```
Comment (lamblin) [inline doc/install.txt]: We should at least keep "If you want the bleedeng edge version" before that command.

File: doc/install.txt [hunk 2]
```diff
@@ -507,62 +507,38 @@ Note that:
...
 MacOS
 -----
 
-There are various ways to install Theano dependencies on a Mac.
-Here we describe the process in detail with EPD, Anaconda or with MacPorts, but if you did
-it differently and it worked, please let us know the details on the
-`theano-users`_ mailing-list, so that we can add alternate instructions
-here.
+There are various ways to install Theano dependencies on a Mac. Here
+we describe the process in detail with Canopy, Anaconda or Homebrew, but
```
Comment (lamblin) [inline doc/install.txt]: The instructions for MacPorts are still there, why remove them from the list?

File: doc/install.txt (hunk 1 above)
Comment (abergeron) [inline doc/install.txt]: Yes.

File: doc/install.txt (hunk 2 above)
Comment (abergeron) [inline doc/install.txt]: I intended to remove it at first, but then decided against it.  I'll re-add it to the list.

**Comment by nouiz** [comment]
I'm ok with this PR, but I let @lamblin merge.

**Pull request merged**